File: app/src/main/java/com/maksk993/cryptoportfolio/presentation/models/AccountAdapter.java
package com.maksk993.cryptoportfolio.presentation.models;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.maksk993.cryptoportfolio.R;
import com.maksk993.cryptoportfolio.domain.models.Account;
import com.maksk993.cryptoportfolio.domain.models.AssetItem;

import java.util.List;

class AccountAdapterJ extends RecyclerView.Adapter<AssetViewHolder> {
    private OnItemClickListener listener;
    private Context context;
    private List<Account> items;

    public interface OnItemClickListener {
        void onItemClick(int position);
    }

    public AccountAdapterJ(Context context, List<Account> items) {
        this.context = context;
        this.items = items;
    }

    public void setOnItemClickListener(OnItemClickListener listener) {
        this.listener = listener;
    }

    @NonNull
    @Override
    public AssetViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new AssetViewHolder(LayoutInflater.from(context).inflate(R.layout.asset_item, parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull AssetViewHolder holder, int position) {
        holder.assetName.setText(items.get(position).getName());
    }

    @Override
    public int getItemCount() {
        return items.size();
    }
}

File: app/src/main/java/com/maksk993/cryptoportfolio/presentation/models/AccountAdapter.kt
package com.maksk993.cryptoportfolio.presentation.models

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.maksk993.cryptoportfolio.R
import com.maksk993.cryptoportfolio.domain.models.Account


class AccountAdapter(val context: Context, private var items : List<Account>): RecyclerView.Adapter<MyViewHolder>() {
    private lateinit var listener: OnItemClickListener

    interface OnItemClickListener {
        fun onItemClick(position: Int)
    }

    fun setOnItemClickListener(listener: OnItemClickListener) {
        this.listener = listener
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        return MyViewHolder(
            LayoutInflater.from(context).inflate(R.layout.recycler_view_item, parent, false)
        )
    }

    override fun getItemCount(): Int = items.size

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        holder.leftName.text = items[position].name
        holder.itemView.setOnClickListener{
            listener.onItemClick(position)
        }
    }
}